### PR TITLE
Select.all can select in an array of parents

### PR DESCRIPTION
**Pull request by fregante**
`select.all('a', select.all('div'))` now is technically possible. In the real world one might need something like:

``` js
var someElements = select.all('.elements').filter(filterUnwanted);
var someLinksInElements = select.all('[href]', someElements);
```

Or:

``` js
var someElements = select.all('a', [parentFoo, parentBar]);
```

I simplified this a lot from what I originally had and I tested it again on a production site.

Thoughts? This is the most I'd ask this lib to do. If it sounds cool I'll update the docs.

### CODE REVIEW

**Comment by azer** [comment]
would this affect the performance of simpler usages ?

**Comment by fregante** [comment]
8% on Chrome: http://jsperf.com/dom-select-array-support-vs-not

If we want to optimize for the most common use case (`parent`-less) I can add one line:

``` js
 if (!parent) return getElementsArray(_select(selector, document));
```

**Comment by fregante** [comment]
I converted it to just use procedural programming instead of functional. It's even faster than the 1.1.0 code in Chrome and only marginally slower in Firefox http://jsperf.com/dom-select-array-support-vs-not/3

**Comment by TehShrike** [comment]
It makes sense to me. Anyone else have any thoughts?  @bfred-it, would you be willing to add some tests for this use case?

**Comment by fregante** [comment]
Sure!

**Comment by TehShrike** [comment]
Sweet! You'll have to rebase to get the new test file, I touched it just earlier today.

**Comment by fregante** [comment]
Ended up publishing this as [select-dom](https://github.com/bfred-it/select-dom)

File: index.js [hunk 1]
```diff
@@ -1,13 +1,31 @@
...
-function all (selector, parent) {
-  parent || (parent = document);
-  var selection = parent.querySelectorAll(selector);
-  return  Array.prototype.slice.call(selection);
-}
+module.exports = function (selector, parent) {
+	return (parent || document).querySelector(selector);
+};
+
+var current, i, l, ii, ll;
```
Comment (TehShrike) [inline index.js]: I'm guessing this is a performance optimization, but I still don't like the idea of global variables.

**Pull request closed without merging**